polars/polars-lazy/polars-plan/src/dsl/function_expr/temporal.rs

```diff
@@ -93,21 +93,45 @@ pub(super) fn temporal_range_dispatch(
     );
     const TO_MS: i64 = SECONDS_IN_DAY * 1000;
 
-    let rng_start = start.to_physical_repr();
-    let rng_stop = stop.to_physical_repr();
-    let dtype = start.dtype();
+    let start_dtype = start.dtype();
 
-    let mut start = rng_start.cast(&DataType::Int64)?;
-    let mut stop = rng_stop.cast(&DataType::Int64)?;
+    // Note: `start` and `stop` have already been cast to their supertype,
+    // so only `start`'s dtype needs to be matched against.
+    let (mut start, mut stop) = match start_dtype {
+        #[cfg(feature = "timezones")]
+        DataType::Datetime(_, Some(_)) => (
+            start
+                .datetime()
+                .unwrap()
+                .replace_time_zone(None, None)?
```

Comment:
if `start` and `stop` have a time zone, I'm removing it, as `date_range_impl` deals with that